client/sftp_client/editor.py

```diff
@@ -1,38 +1,35 @@
-import os
 import logging
 
 from . import sftp_utilities
 
 from king_phisher.client.widget import completion_providers
 from king_phisher import utilities
-from king_phisher.client import gui_utilities
 
 from gi.repository import GtkSource
 
-logger = logging.getLogger('KingPhisher.Plugins.SFTPClient.editor')
+logger = logging.getLogger('KingPhisher.Plugins.SFTPClient')
 
-class SftpEditor(object):
+class SFTPEditor(object):
 	"""
 	Handles the editor tab functions
 	"""
-	def __init__(self, file_contents, file_path, location):
+	def __init__(self, file_path, directory, location):
 		"""
 		This class is used to set up the Gtk.SourceView instance to edit the file
 
 		:param str file_contents: 
-		:param file_path: the path of the file to edit
-		:param location: the locate either remote or local
+		:param str file_path: the path of the file to edit
+		:param directory: the local or remote directory instance
+		:param str location: the locate either remote or local
```

Comment:
Location should not have to be specified. If you need the string representation of it, please add a property to the directory. "local" for LocalDirectory "unknown" for the base class etc, but this should not have to be hardcoded in where this class is initialized.